zerver/actions/create_user.py

```diff
@@ -606,6 +606,26 @@ def do_reactivate_user(user_profile: UserProfile, *, acting_user: Optional[UserP
             }
         ).decode(),
     )
+
+    bot_owner_changed = False
+    if (
+        user_profile.is_bot
+        and user_profile.bot_owner is not None
+        and not user_profile.bot_owner.is_active
+        and acting_user is not None
+    ):
+        previous_owner = user_profile.bot_owner
+        user_profile.bot_owner = acting_user
+        user_profile.save()  # Can't use update_fields because of how the foreign key works.
+        RealmAuditLog.objects.create(
+            realm=user_profile.realm,
+            acting_user=acting_user,
+            modified_user=user_profile,
+            event_type=RealmAuditLog.USER_BOT_OWNER_CHANGED,
+            event_time=event_time,
```

Comment:
Do we want `extra_data` to flag that this was during reactivation? Maybe it's not necessary.

Reply:
We already would have `RealmAuditLog` entry for reactivation just before this, so might not be necessary to include this in `extra_data`.

Reply:
OK that seems reasonable.